dao/src/main/java/org/thingsboard/server/dao/ai/AiModelSettingsCacheEvictEvent.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright © 2016-2025 The Thingsboard Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.thingsboard.server.dao.ai;
+
+import org.thingsboard.server.common.data.id.AiModelSettingsId;
+import org.thingsboard.server.common.data.id.TenantId;
+
+import java.util.Set;
+
+record AiModelSettingsCacheEvictEvent(Set<AiModelSettingsCacheKey> keys) {
+
+    static AiModelSettingsCacheEvictEvent of(TenantId tenantId, AiModelSettingsId settingsId) {
```

Comment:
Make sure everyone passes the correct tenant id (e.g., not sys tenant id), or only use id for the key, without tenant id

Reply:
DataValidator already validates this for the AI model settings, but I’ll add additional checks.